### PR TITLE
fix: chinese garbled characters #1617

### LINKED ISSUE #1617: AppiumDriverLocalService.addLogMessageConsumer(Consumer<String>) supply garbled  characters in chinese words , 中文字符乱码
## Description

`AppiumDriverLocalService` supply garbled characters in chinese words  when call the method `addLogMessageConsumer`

## Environment

* Java client build version or git revision if you use some snapshot:  java-client-v7.5.1.jar
* Appium server version or git revision if you use some snapshot: 1.22.1
* Desktop OS/version used to run Appium if necessary: win10
* Node.js version (unless using Appium.app|exe) or Appium CLI or Appium.app|exe: v16.13.0
* Mobile platform/version under test: android8/11
* Real device or emulator/simulator: UIAutomator2

## Code To Reproduce Issue [ Good To Have ]
when you use `driver.findElementByXPath("//*[@text='发现' ]")` , your log got this `@text='￥ﾏﾑ￧ﾎﾰ'`
image

```
        final AppiumServiceBuilder warn = new AppiumServiceBuilder().usingPort(port).withArgument(GeneralServerFlag.LOG_LEVEL, "info");
        final AppiumDriverLocalService appiumDriverLocalService = AppiumDriverLocalService.buildService(warn);
        appiumDriverLocalService.addLogMessageConsumer(s -> log.info("appium log: {}", s));
        appiumDriverLocalService.addLogMessageConsumer(s -> System.out.println(s));
        appiumDriverLocalService.start();
```

## Exception Stacktraces

Please create a [gist](https://gist.github.com) with the pasted stacktrace of the exception thrown by java.

## Link To Appium Logs

Please create a [gist](https://gist.github.com) which is a paste of your _full_ Appium logs, and link them here. Do _not_ paste your full Appium logs here, as it will make this issue very long and hard to read! 
If you are reporting a bug, _always_ include Appium logs as linked gists! It helps to define the problem correctly and clearly.

### PR DESCRIPTION
**Pull request by 819635822**
fix  #1617
## Change list

avoid chinese garbled characters 
 
## Types of changes

What types of changes are you proposing/introducing to Java client?
_Put an `x` in the boxes that apply_

- [ ] No changes in production code.
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Details

details  #1617

### CODE REVIEW

File: src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java [hunk 1]
```diff
@@ -410,16 +411,15 @@ static Slf4jLogMessageContext parseSlf4jContextFromLogMessage(String logMessage)
...
             @Override
             public void write(int chr) {
                 try {
-                    lineBuilder.append((char) chr);
-                    Matcher matcher = LOG_MESSAGE_PATTERN.matcher(lineBuilder.toString());
-                    if (matcher.matches()) {
-                        consumer.accept(matcher.group(1));
-                        lineBuilder = new StringBuilder();
+                    outputStream.write(chr);
+                    if ((chr == '\n')) {
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java:420]: double brackets are not needed

File: src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java [hunk 2]
```diff
@@ -410,16 +411,15 @@ static Slf4jLogMessageContext parseSlf4jContextFromLogMessage(String logMessage)
...
+                        consumer.accept(outputStream.toString());
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java:421]: should we explicitly provide the encoding here?

Comment (819635822) [reply src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java:421]: default charset is good for me

File: src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java (hunk 1 above)
Comment (819635822) [reply src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java]: already fixed

File: src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java [hunk 3]
```diff
@@ -410,16 +411,15 @@ static Slf4jLogMessageContext parseSlf4jContextFromLogMessage(String logMessage)
     public void addLogMessageConsumer(Consumer<String> consumer) {
         checkNotNull(consumer, "consumer parameter is NULL!");
         addOutPutStream(new OutputStream() {
-            StringBuilder lineBuilder = new StringBuilder();
+            private ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java:414]: should the property be `final`?

**Comment by mykola-mokhnach** [comment]
Could you please sign the CLA?

File: src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java (hunk 3 above)
Comment (819635822) [reply src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java:414]: yes , `final` is better

**Comment by 819635822** [comment]
> Could you please sign the CLA?

**Comment by 819635822** [comment]
will CLA affect my pr?

**Comment by mykola-mokhnach** [comment]
I cannot merge it without CLA being signed

**Comment by 819635822** [comment]
👌i will do this later

File: src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java (hunk 2 above)
Comment (valfirst) [reply src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java:421]: it makes sense to introduce the ability to rely on default charset or use a custom one

**Review by mykola-mokhnach — APPROVED** [review]

**Comment by rishikhanna1401** [comment]
@mykola-mokhnach I am facing issue with latest java-client to launch appium service. Does anything changed around this?

While trying to programmatically open appium server using java-client 8.0.0 I see below error msg:

io.appium.java_client.service.local.AppiumServerHasNotBeenStartedLocallyException: The local appium server has not been started. The given Node.js executable: /usr/local/bin/appium Arguments: [/usr/local/lib/node_modules/appium/lib/appium.js, --port, 18384, --address, 0.0.0.0]
Process output: usage: /usr/local/lib/node_modules/appium/build/lib/main.js
[-h] [–shell] [–allow-cors] [–reboot] [–ipa IPA] [-a ADDRESS]
[-p PORT] [-pa BASEPATH] [-ka KEEPALIVETIMEOUT] [-ca CALLBACKADDRESS]
[-cp CALLBACKPORT] [-bp BOOTSTRAPPORT] [-r BACKENDRETRIES]
[–session-override] [-l] [-g LOGFILE]
[–log-level {info,info:debug,info:info,info:warn,info:error,warn,warn:debug,warn:info,warn:warn,warn:error,error,error:debug,error:info,error:warn,error:error,debug,debug:debug,debug:info,debug:warn,debug:error}]
[–log-timestamp] [–local-timezone] [–log-no-colors] [-G WEBHOOK]
[–safari] [–default-device] [–force-iphone] [–force-ipad]
[–tracetemplate AUTOMATIONTRACETEMPLATEPATH]
[–instruments INSTRUMENTSPATH] [–nodeconfig NODECONFIG]
[-ra ROBOTADDRESS] [-rp ROBOTPORT]
[–chromedriver-executable CHROMEDRIVEREXECUTABLE] [–show-config]
[–no-perms-check] [–strict-caps] [–isolate-sim-device]
[–tmp TMPDIR] [–trace-dir TRACEDIR] [–debug-log-spacing]
[–suppress-adb-kill-server] [–long-stacktrace]
[–webkit-debug-proxy-port WEBKITDEBUGPROXYPORT]
[–webdriveragent-port WDALOCALPORT] [-dc DEFAULTCAPABILITIES]
[–relaxed-security] [–allow-insecure ALLOWINSECURE]
[–deny-insecure DENYINSECURE]
[–command-timeout DEFAULTCOMMANDTIMEOUT] [-k]
[–platform-name PLATFORMNAME] [–platform-version PLATFORMVERSION]
[–automation-name AUTOMATIONNAME] [–device-name DEVICENAME]
[–browser-name BROWSERNAME] [–app APP] [-lt LAUNCHTIMEOUT]
[–language LANGUAGE] [–locale LOCALE] [-U UDID]
[–orientation ORIENTATION] [–no-reset] [–full-reset]
[–app-pkg APPPACKAGE] [–app-activity APPACTIVITY]
[–app-wait-package APPWAITPACKAGE]
[–app-wait-activity APPWAITACTIVITY]
[–device-ready-timeout DEVICEREADYTIMEOUT]
[–android-coverage ANDROIDCOVERAGE] [–avd AVD] [–avd-args AVDARGS]
[–use-keystore] [–keystore-path KEYSTOREPATH]
[–keystore-password KEYSTOREPASSWORD] [–key-alias KEYALIAS]
[–key-password KEYPASSWORD] [–intent-action INTENTACTION]
[–intent-category INTENTCATEGORY] [–intent-flags INTENTFLAGS]
[–intent-args OPTIONALINTENTARGUMENTS] [–dont-stop-app-on-reset]
[–calendar-format CALENDARFORMAT] [–native-instruments-lib]
[–keep-keychains] [–localizable-strings-dir LOCALIZABLESTRINGSDIR]
[–show-ios-log] [–async-trace] [–chromedriver-port CHROMEDRIVERPORT]
[–log-filters LOGFILTERS] [-v]
/usr/local/lib/node_modules/appium/build/lib/main.js: error: unrecognized arguments: /usr/local/lib/node_modules/appium/lib/appium.js

Sample Start-up code(worked on java-client 8.0.0-beta version:

service = AppiumDriverLocalService
.buildService(new AppiumServiceBuilder()
.usingDriverExecutable(new File("/usr/local/bin/appium"))
.withAppiumJS(
new File(
“/usr/local/lib/node_modules/appium/lib/appium.js”))
.usingPort(port));
service.start();

Appium Version Used: 1.20.0 and 1.22.0

**Pull request merged**